src/System.Web.OData/OData/DefaultContainerBuilder.cs

```diff
@@ -78,7 +79,7 @@ public class DefaultContainerBuilder : IContainerBuilder
         /// <returns>The container built by this builder.</returns>
         public virtual IServiceProvider BuildContainer()
         {
-            return services.BuildServiceProvider();
+            return (IServiceProvider)services.GetType().GetTypeInfo().Assembly.GetType(typeof(ServiceCollectionContainerBuilderExtensions).GetTypeInfo().FullName).GetMethod(nameof(ServiceCollectionContainerBuilderExtensions.BuildServiceProvider), new[] { typeof(IServiceCollection) }).Invoke(null, new object[] { services });
```

Comment:
I like the approach but the code is a bit unreadable. Can you break it up on multiple lines?

Also, can you add some comments? Without an explanation, someone is likely to wonder why we go through all this trouble instead of calling the method directly.